### DISCUSSION TITLE
How to store images as blobs and retrieve them from the database as an image file.

### DISCUSSION THREAD

**Discussion by NicholusMuwonge** [supabase/supabase#7212] [Questions]
I have a table that accepts `bytea` data type to be able to store an image. I have been able to save the image to the database and in the table it looks like this on the photo field `{"type":"Buffer","data":[123,34,116,121,112,101,34,58,34,66,117,102,102,101,114,34,44,34,100,97,116,97,34,58,91,49,50,51,44,51,52,44,52,56,44,51,52,44,53,56,44,49,50,51,44,49,50,53,44,49,50,53,93,125]}` but when I retrive the data on a get call, I get sent back something like this on the photo field `\x7b2274797065223a22427566666572222c2264617461223a5b3132332c33342c34382c33342c35382c3132332c3132352c3132355d7d`. 

I need some help understanding how I can get back the image I sent to the database in JavaScript.

Post call looks like 
```
const { data, error, status } = await supabase.from("some_table").insert([
      {
        field1: val,
        field2: val2,
        photo: imagefile, ---> This is an image file
      },
    ]);
```

I am using React/Next.js

**Comment by nattstack** [comment]
I don't know why reading the `bytea` data comes out like that. I have a solution for storing JavaScript objects.

```js
const buffObj = {"type":"Buffer","data":[123,34,116,121,112,101,34,58,34,66,117,102,102,101,114,34,44,34,100,97,116,97,34,58,91,49,50,51,44,51,52,44,52,56,44,51,52,44,53,56,44,49,50,51,44,49,50,53,44,49,50,53,93,125]};

const { data, error, status } = await supabase.from("some_table").insert([
      {
        field1: val,
        field2: val2,
        photo: JSON.stringify(buffObj)
      },
    ]);
```

and on your table, change column to `varchar` or `text`.

**Reply by NicholusMuwonge** [reply]
Thank you for the suggestion but I had decided to go ahead and work with the buckets and save the string instead.
